src/main/java/lotto/service/LottoService.java

```diff
@@ -0,0 +1,129 @@
+package lotto.service;
+
+import java.util.List;
+import java.util.Objects;
+import lotto.dto.LottoDto;
+import lotto.dto.LottoPurchaseDto;
+import lotto.dto.LottosDto;
+import lotto.dto.ReturnRateDto;
+import lotto.dto.WinningResultDto;
+import lotto.model.Lotto;
+import lotto.model.LottoPurchaseAmount;
+import lotto.model.LottoPurchaseQuantity;
+import lotto.model.Lottos;
+import lotto.model.ReturnRate;
+import lotto.model.WinningResult;
+
+public class LottoService implements Service {
+    private LottoService() {
+    }
+
+    public static LottoService create() {
+        return new LottoService();
+    }
+
+    @Override
+    public LottoPurchaseDto buyLottery(final String inputPurchaseAmount) {
+        Objects.requireNonNull(inputPurchaseAmount);
+        LottoPurchaseAmount amount = LottoPurchaseAmount.from(inputPurchaseAmount);
+        LottoPurchaseQuantity quantity = amount.getLottoPurchaseQuantity();
+        return getLottoPurchaseDto(amount, quantity);
+    }
+
+    private LottoPurchaseDto getLottoPurchaseDto(
+            final LottoPurchaseAmount amount,
+            final LottoPurchaseQuantity quantity
+    ) {
+        Objects.requireNonNull(amount);
+        Objects.requireNonNull(quantity);
+        return new LottoPurchaseDto(amount.getAmount(), quantity.quantity());
+    }
+
+    @Override
+    public LottosDto generateLotto(final int quantity) {
+        return getLottosDto(Lottos.from(quantity));
+    }
+
+    private LottosDto getLottosDto(final Lottos lottos) {
+        Objects.requireNonNull(lottos);
+        List<LottoDto> lottoDtos = lottos.getLotto()
+                .stream()
+                .map(this::getLottoDto)
+                .toList();
+        return new LottosDto(lottoDtos);
```

Comment:
고민하시던 서비스에 위치한 생성로직이 이런 형태이군요 🤔 
서비스 레이어에서 DTO를 반환하는 것은 제 눈엔 굉장히 타당해보이긴 한데 DTO를 만들기 위해서 서비스 레이어가 위와 같은 형태로 작동하는 것은 다른 방식이 더 좋지 않나라는 생각이 들어요 

예를 들면 DTO 클래스에 정적 팩토리 메서드를 만들어서 

```suggestion
        return LottosDto.from(lottos)
```
와 같이 생성로직을 DTO 내부에 위치시키면 서비스 레이어의 책임이 줄어들 수 있지 않은가 하는 개인적인 생각입니다 😄

Reply:
`return new LottosDto(lottoDtos);`와 같이 사용했던 건 LottosDto는 레코드인데 레코드의  생성자는 항상 public이어서 정적 팩토리 메서드를 사용할 경우 생성 방법이 new와 정팩메 두 가지가 생겨서 굳이 정팩메를 만들어야 하나 하는 생각이 들었습니다. 이부분에 대해서 저도 좀 더 파헤쳐 봐야할 것 같습니다. 좋은 말씀 감사합니다. 😀